src/commands.ts

```diff
@@ -137,6 +124,35 @@ export async function exportPluginFromContextMenu(uri: vscode.Uri): Promise<void
     fileutils.storeFileExtensionConfig(path.dirname(scriptPath), extensions);
 }
 
+export async function registerPlugin(uri: vscode.Uri): Promise<void> {
+    const scriptPath: string = uri.fsPath;
+    const pluginName: string = path.basename(path.dirname(scriptPath));
+    const exportPath: string = path.join(os.tmpdir(), `${pluginName}.uri`);
+    const extensions = await readOrRequestFileExtensionConfig(uri);
+    if (!extensions) {
+        return;
+    }
+
+    await vscu.exportDataPlugin(scriptPath, extensions.toString(), `${exportPath}`, false);
+    const process = await open(exportPath);
+    process.on('exit', rc => {
+        const foundUsiReg = rc === 0;
+        if (!foundUsiReg) {
```

Comment:
What happens if the Plugin is already registered? Is the currently registered version overwritten silently?

Reply:
It actually behaves like a double-click on the uri file. It shows the usiReg.exe dialog, prompting whether you really wish to overwrite the existing Plugin. It's not done silently.